tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/TaskParams.java

```diff
@@ -96,6 +106,56 @@ public String searchBase() {
         return variables.assertString(LDAP_SEARCH_BASE);
     }
 
+    @Override
+    public boolean certificate() {
+        Map<String, Object> certificate = variables.getMap(LDAP_CERTIFICATE, null);
+
+        if (certificate == null || certificate.isEmpty()) {
+            return false;
+        }
+
+        Path certPath = Paths.get(CustomSocketFactory.CERT_PATH);
+
+        cleanExistingCerts(certPath);
+        copyCertFile((String) certificate.get("file"), certPath);
+        writeCertString((String) certificate.get("text"), certPath);
+        // TODO support secret source?
```

Comment:
can we move all this file operations outside of TaskParams?
TaskParams is just wrapper on `variables` with type validation.
it is unexpected for me that the method copies something somewhere every time :)

what if split this method to:
```
public Path certificateFile() {
        Map<String, Object> certificate = variables.getMap(LDAP_CERTIFICATE, null);

        if (certificate == null || certificate.isEmpty()) {
            return null;
        }

        String file = MapUtils.getString(certificate, "file"));
        if (file == null) {
             return null;
        }

        Path result = Paths.get(file);
        if (Files.notExists(result)) {
           throw error ...
        }
        return result;
}
```
and 
```
public String certificateText() {
...
}
```

and then copy/create files in task.
also should we throw exception/warning if user specify text and file?

Reply:
The quirk I was working against is that the custom `SocketFactory` class need to have the certificate file (one or more certs in one pem file) available in a [pre-defined location](https://github.com/walmartlabs/concord-plugins/blob/6032fa630da5235df56a18dc94a1673da7aedc8b/tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/CustomSocketFactory.java#L49) because we [don't instantiate it directly](https://github.com/walmartlabs/concord-plugins/blob/6032fa630da5235df56a18dc94a1673da7aedc8b/tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/LdapTaskCommon.java#L225).

I get what you mean about it being weird to have a method returning a `boolean` but it's creating files every time you call it (even though it's only called once...at the moment). I'll move those elsewhere (`LdapTaskCommon` probably).

> also should we throw exception/warning if user specify text and file?

I may be over-engineering for this one, but it _could_ be possible to have some project config-defined cert text along with a `concord.yml`-defined extra certs file. Anyway yeah, that's likely to never actually happen and they're just public certs so copying them into a single input parameter isn't that crazy of a requirement.

Reply:
Addressed in next commit.

```
Custom certificate may only be given exclusively as 'text' or 'file'
```